src/services/navigationBar.ts

```diff
@@ -2,856 +2,552 @@
-                            break;
-                        }
-                        // Fall through
-                    case SyntaxKind.ClassDeclaration:
-                    case SyntaxKind.EnumDeclaration:
-                    case SyntaxKind.InterfaceDeclaration:
-                    case SyntaxKind.ModuleDeclaration:
-                    case SyntaxKind.FunctionDeclaration:
-                    case SyntaxKind.ImportEqualsDeclaration:
-                    case SyntaxKind.ImportSpecifier:
-                    case SyntaxKind.ExportSpecifier:
-                    case SyntaxKind.TypeAliasDeclaration:
-                        childNodes.push(node);
-                        break;
-                }
-            }
-
-            // for (let i = 0, n = nodes.length; i < n; i++) {
-            //    let node = nodes[i];
-
-            //    if (node.kind === SyntaxKind.ClassDeclaration ||
-            //        node.kind === SyntaxKind.EnumDeclaration ||
-            //        node.kind === SyntaxKind.InterfaceDeclaration ||
-            //        node.kind === SyntaxKind.ModuleDeclaration ||
-            //        node.kind === SyntaxKind.FunctionDeclaration) {
-
-            //        childNodes.push(node);
-            //    }
-            //    else if (node.kind === SyntaxKind.VariableStatement) {
-            //        childNodes.push.apply(childNodes, (<VariableStatement>node).declarations);
-            //    }
-            // }
-            forEach(nodes, visit);
-            return sortNodes(childNodes);
-        }
-
-        function getTopLevelNodes(node: SourceFile): Node[] {
-            const topLevelNodes: Node[] = [];
-            topLevelNodes.push(node);
-
-            addTopLevelNodes(node.statements, topLevelNodes);
-
-            return topLevelNodes;
-        }
+    export function getNavigationBarItems(sourceFile: SourceFile): NavigationBarItem[] {
+        const root = createNavNode(undefined, sourceFile);
+        return map(topLevelItems(root), convertToTopLevelItem);
+    }
 
-        function sortNodes(nodes: Node[]): Node[] {
-            return nodes.slice(0).sort((n1: Declaration, n2: Declaration) => {
-                if (n1.name && n2.name) {
-                    return localeCompareFix(getPropertyNameForPropertyNameNode(n1.name), getPropertyNameForPropertyNameNode(n2.name));
-                }
-                else if (n1.name) {
-                    return 1;
-                }
-                else if (n2.name) {
-                    return -1;
-                }
-                else {
-                    return n1.kind - n2.kind;
-                }
-            });
+    /**
+     * Represents a navBar item and its children.
+     * The returned NavigationBarItem is more complicated and doesn't include 'parent', so we use these to do work before converting.
+     */
+    interface NavNode {
+        node: Node;
+        additionalNodes: Node[]; // May be missing

```

Comment:
If the comment is true, I'd remove the comment, and write `additionalNodes?: Node[];` instead. (This would likely be caught if we could compile the compiler under strict null checks)